File: temp_groq_probe.py
import json
import urllib.request
import urllib.error
from pathlib import Path

p = Path('.env.local')
text = p.read_bytes()
if text.startswith(b'\xff\xfe') or text.startswith(b'\xfe\xff'):
    text = text.decode('utf-16')
else:
    text = text.decode('utf-8')
values = {}
for line in text.splitlines():
    line = line.strip()
    if not line or line.startswith('#'):
        continue
    if '=' in line:
        k, v = line.split('=', 1)
        k = k.strip()
        v = v.strip()
        if v and v[0] in '"\'' and v[-1] == v[0]:
            v = v[1:-1]
        values[k] = v

body = json.dumps({
    'model': 'llama-3.3-70b-versatile',
    'messages': [
        {'role': 'system', 'content': 'You are a helpful assistant.'},
        {'role': 'user', 'content': 'Return a compact JSON object with a single field hello.'}
    ],
    'temperature': 0.1,
    'response_format': {'type': 'json_object'}
}).encode()
req = urllib.request.Request('https://api.groq.com/openai/v1/chat/completions', data=body, headers={'Content-Type': 'application/json', 'Authorization': f"Bearer {values['GROQ_API_KEY']}"}, method='POST')
try:
    with urllib.request.urlopen(req, timeout=45) as resp:
        print('status', resp.status)
        print(resp.read().decode())
except urllib.error.HTTPError as e:
    print('status', e.code)
    print(e.read().decode())
except Exception as e:
    print('err', repr(e))

File: temp_groq_probe.mjs
import fs from 'node:fs';
import path from 'node:path';

const filePath = path.resolve('.env.local');
const raw = fs.readFileSync(filePath);
const text = raw[0] === 0xff && raw[1] === 0xfe || raw[0] === 0xfe && raw[1] === 0xff ?
    raw.toString('utf16le') :
    raw.toString('utf8');
const values = {};
for (const line of text.split(/\r?\n/)) {
    const trimmed = line.trim();
    if (!trimmed || trimmed.startsWith('#') || !trimmed.includes('=')) continue;
    const [k, ...rest] = trimmed.split('=');
    let v = rest.join('=').trim();
    if ((v.startsWith('"') && v.endsWith('"')) || (v.startsWith("'") && v.endsWith("'"))) {
        v = v.slice(1, -1);
    }
    values[k.trim()] = v;
}

const body = JSON.stringify({
    model: 'llama-3.3-70b-versatile',
    messages: [
        { role: 'system', content: 'You are a helpful assistant.' },
        { role: 'user', content: 'Return a compact JSON object with a single field hello.' },
    ],
    temperature: 0.1,
    response_format: { type: 'json_object' },
});

const res = await fetch('https://api.groq.com/openai/v1/chat/completions', {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${values.GROQ_API_KEY}`,
    },
    body,
});
console.log('status', res.status);
console.log(await res.text());